diff --git a/ap.c b/ap.c
index 9fb9b10..b0860a1 100644
--- a/ap.c
+++ b/ap.c
@@ -8202,6 +8202,11 @@ write_conf:
 			fprintf(f, "ieee80211d=1\n");
 			fprintf(f, "ieee80211h=1\n");
 		}
+	        else if (dut->ap_channel >= 52 && dut->ap_channel <= 144) {
+                         fprintf(f, "country_code=US\n"
+                                "ieee80211d=1\n"
+                                "ieee80211h=1\n");
+                }
 	}
 
 	if (drv == DRIVER_LINUX_WCN && dut->ap_mode == AP_11ax) {
