diff --git a/ap.c b/ap.c
index b9b5ecf..f1b6d4d 100644
--- a/ap.c
+++ b/ap.c
@@ -8164,10 +8164,11 @@ write_conf:
 			ht40plus = 0;
 		}
 
-		fprintf(f, "ht_capab=%s%s%s\n",
+		fprintf(f, "ht_capab=%s%s%s%s\n",
+			"[SHORT-GI-20][SHORT-GI-40]",
 			ht40plus ? "[HT40+]" : "",
 			ht40minus ? "[HT40-]" : "",
-			tx_stbc ? "[TX-STBC]" : "");
+			tx_stbc ? "[TX-STBC][RX-STBC1]" : "");
 	}
 
 	if ((drv == DRIVER_MAC80211 || drv == DRIVER_QNXNTO ||
@@ -8899,7 +8900,7 @@ skip_key_mgmt:
 				(dut->ap_ldpc == VALUE_ENABLED) ?
 				"[RXLDPC]" : "",
 				(dut->ap_tx_stbc == VALUE_ENABLED) ?
-				"[TX-STBC-2BY1]" : "",
+				"[TX-STBC-2BY1][RX-STBC-1]" : "",
 				dut->ap_mu_txBF ? "[MU-BEAMFORMER]" : "",
 				dut->ap_chwidth == AP_160 ? "[VHT160]" :
 				(dut->ap_chwidth == AP_80_80 ?
