--- a/ap.c
+++ b/ap.c
@@ -1117,6 +1117,10 @@ static enum sigma_cmd_result cmd_ap_set_wireless(struct sigma_dut *dut,
 		}
 	}
 
+	val = get_param(cmd, "MU_TxBF");
+	if (val)
+		dut->ap_mu_txBF = strcasecmp(val, "enable") == 0;
+
 	/* UNSUPPORTED: tx_lgi_rate */
 
 	val = get_param(cmd, "wpsnfc");
@@ -7684,7 +7688,7 @@ skip_key_mgmt:
 				"[SU-BEAMFORMER][SU-BEAMFORMEE][BF-ANTENNA-2][SOUNDING-DIMENSION-2]" : "",
 				(dut->ap_ldpc == VALUE_ENABLED) ?
 				"[RXLDPC]" : "",
-				dut->ap_tx_stbc ? "[TX-STBC-2BY1]" : "",
+				dut->ap_tx_stbc ? "[TX-STBC-2BY1][RX-STBC-1]" : "",
 				dut->ap_mu_txBF ? "[MU-BEAMFORMER]" : "");
 		}
 	}
