diff --git a/ap.c b/ap.c
index b9b5ecf..1e3d61d 100644
--- a/ap.c
+++ b/ap.c
@@ -1341,6 +1341,10 @@ static enum sigma_cmd_result cmd_ap_set_wireless(struct sigma_dut *dut,
 		}
 	}
 
+	val = get_param(cmd, "MU_TxBF");
+	if (val)
+		dut->ap_mu_txBF = strcasecmp(val, "enable") == 0;
+
 	/* UNSUPPORTED: tx_lgi_rate */
 
 	val = get_param(cmd, "wpsnfc");
@@ -8899,7 +8903,7 @@ skip_key_mgmt:
 				(dut->ap_ldpc == VALUE_ENABLED) ?
 				"[RXLDPC]" : "",
 				(dut->ap_tx_stbc == VALUE_ENABLED) ?
-				"[TX-STBC-2BY1]" : "",
+				"[TX-STBC-2BY1][RX-STBC-1]" : "",
 				dut->ap_mu_txBF ? "[MU-BEAMFORMER]" : "",
 				dut->ap_chwidth == AP_160 ? "[VHT160]" :
 				(dut->ap_chwidth == AP_80_80 ?
