diff --git a/ap.c b/ap.c
index ba2a57b..053c991 100644
--- a/ap.c
+++ b/ap.c
@@ -8171,7 +8171,7 @@ write_conf:
 		fprintf(f, "ht_capab=%s%s%s\n",
 			ht40plus ? "[HT40+]" : "",
 			ht40minus ? "[HT40-]" : "",
-			tx_stbc ? "[TX-STBC]" : "");
+			tx_stbc ? "[TX-STBC][RX-STBC1]" : "[RX-STBC1]");
 	}
 
 	if ((drv == DRIVER_MAC80211 || drv == DRIVER_QNXNTO ||
@@ -8903,7 +8903,7 @@ skip_key_mgmt:
 				(dut->ap_ldpc == VALUE_ENABLED) ?
 				"[RXLDPC]" : "",
 				(dut->ap_tx_stbc == VALUE_ENABLED) ?
-				"[TX-STBC-2BY1][RX-STBC-1]" : "",
+				"[TX-STBC-2BY1][RX-STBC-1]" : "[RX-STBC-1]",
 				dut->ap_mu_txBF ? "[MU-BEAMFORMER]" : "",
 				dut->ap_chwidth == AP_160 ? "[VHT160]" :
 				(dut->ap_chwidth == AP_80_80 ?
@@ -10021,7 +10021,7 @@ static enum sigma_cmd_result cmd_ap_reset_default(struct sigma_dut *dut,
 		 * is still not set, query it from the wiphy to see if it is
 		 * supported by the driver.
 		 */
-		if (drv == DRIVER_LINUX_WCN)
+		if (drv == DRIVER_LINUX_WCN || drv == DRIVER_MAC80211)
 			dut->ap_tx_stbc = VALUE_NOT_SET;
 		else
 			dut->ap_tx_stbc = VALUE_ENABLED;
